test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,192 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+public class AddCommandTest {
+	private static final List<? extends ReadOnlyPerson> LAST_SHOWN_LIST = Collections.emptyList();
+	
+	/*
+	 * Note how the names of the test methods does not follow the normal naming
+	 * convention. That is because our coding standard allows a different naming
+	 * convention for test methods.
```

Comment:
@Neurrone can remove this comment. No need to repeat it in every class. In fact, we should move it out of the code and mention it once in a more central location, but that's a separate issue.